rustls/src/client/tls12.rs

```diff
@@ -387,9 +387,7 @@ impl State<ClientConnectionData> for ExpectServerKx {
             })?;
 
         // Save the signature and signed parameters for later verification.
-        let mut kx_params = Vec::new();
-        ecdhe.params.encode(&mut kx_params);
-        let server_kx = ServerKxDetails::new(kx_params, ecdhe.dss);
+        let server_kx = ServerKxDetails::new(ecdhe.params.get_encoding(), ecdhe.dss);
```

Comment:
Would be nice to land these changes separately (either in a separate commit if you're going to squash and clean up commits here or in a separate PR).

Reply:
Same answer as https://github.com/rustls/rustls/pull/1533#discussion_r1357165250